docs/source/training.rst

```diff
@@ -0,0 +1,276 @@
+meaning that you can use them just as you would any model in PyTorch for
+both inference and optimization.
+
+Let's consider the common task of fine-tuning a masked language model like
+BERT on a sequence classification task. When we instantiate a model with
+:func:`~transformers.PreTrainedModel.from_pretrained`, the model
+configuration and pre-trained weights
+of the specified model are used to initialize the model. The
+library also includes a number of task-specific final layers or 'heads' whose
+weights are instantiated randomly when not present in the specified
+pre-trained model. For example, instantiating a model with
+``BertForSequenceClassification.from_pretrained('bert-base-uncased', num_classes=2)``
+will create a BERT model instance with encoder weights copied from the
+``bert-base-uncased`` model and a randomly initialized sequence
+classification head on top of the encoder with an output size of 2.
+
+This is useful because it allows us to make use of the pre-trained BERT
+encoder and easily train it on whatever sequence classification dataset we
+choose. Models are initialized in ``eval`` mode by default. To train, be sure
+to first call ``model.train()``. When we call the model with the target labels
+as a tensor, the first returned element is the Cross Entropy loss between the
+classifier predictions and the passed labels. The following demonstrates
+how you could instantiate a pre-trained model and perform a single update on
+a dummy sentiment classification batch:
+
+.. code-block:: python
+
+    from transformers import BertTokenizer, BertForSequenceClassification, AdamW
+
+    tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+    model = BertForSequenceClassification.from_pretrained('bert-base-uncased')
+    model.train()
+    optimizer = AdamW(model.parameters(), lr=5e-5) # any PT optimizer works
+
+    text_batch = ["I love Pixar.", "I don't care for Pixar."]
+    labels = torch.tensor([1,0]).unsqueeze(0)
+    encoding = tokenizer.batch_encode_plus(text_batch, return_tensors='pt',
+                                           pad_to_max_length=True)
+    input_ids = encoding['input_ids']
+    attention_mask = encoding['attention_mask']
+
+    outputs = model(input_ids, attention_mask=attention_mask, labels=labels)
+    loss = outputs[0]
+    loss.backward()
+    optimizer.step()
+
+Of course, you can train on GPU by calling ``to('cuda')`` on the model and
+inputs as usual.
+
+We also provide a few tools to make it easier to apply more complex
+fine-tuning procedures. Here's an example using weight decay and learning
+rate scheduling (omitting the data preparation steps for brevity):
+
+.. code-block:: python
+
+    from transformers import get_linear_schedule_with_warmup
+
+    device = torch.device('cuda')
+    model = BertForSequenceClassification.from_pretrained('bert-base-uncased').to(device)
+    model.train()
+
+    # Prepare optimizer and schedule (linear warmup and decay)
```

Comment:
Same as before, consider splitting the code at each commented line and introduce in prose what happens after.